spec/policies/user_policy_spec.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe UserPolicy do
+  let(:user) { create :user, :admin }
+  let(:learning_partner) { create :learning_partner }
+  let(:parent_team) { create :team, learning_partner: }
+  let(:sub_team) { create :team, parent_team:, learning_partner: }
+
+  describe '#change_team?' do
+    before do
+      @learner = create(:user, :learner, team: sub_team, learning_partner:)
+      @manager = create(:user, :manager, team: sub_team, learning_partner:)
+      @owner = create(:user, :owner, team: parent_team, learning_partner:)
+    end
+
+    it 'returns false if user is not privileged user' do
+      expect(described_class.new(@learner, @manager)).not_to be_change_team
+    end
+
+    it 'returns false if changing own team' do
+      expect(described_class.new(@manager, @manager)).not_to be_change_team
+    end
+
+    it 'returns false if changing owner' do
+      new_owner = create(:user, :owner, team: parent_team, learning_partner:)
+
+      expect(described_class.new(new_owner, @owner)).not_to be_change_team
+    end
+
+    it 'returns false if manager trying to change team of another manager belongs to same team' do
+      new_manager = create(:user, :manager, team: sub_team, learning_partner:)
+      expect(described_class.new(@manager, new_manager)).not_to be_change_team
+    end
+
+    it 'returns false if sub team manager trying to change team of parent team manager' do
+      new_manager = create(:user, :manager, team: parent_team, learning_partner:)
+      expect(described_class.new(@manager, new_manager)).not_to be_change_team
+    end
+
+    it 'manager can change team of any manager in its sub teams' do
+      new_manager = create(:user, :manager, team: parent_team, learning_partner:)
+      expect(described_class.new(new_manager, @manager)).to be_change_team
+    end
+
+    it 'manager can change team of any learner' do
```

Comment:
manager can change team of any learner in his team or his sub teams. He can't change learners in his parent team or above.